includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -188,16 +189,28 @@ public function update_domain_association_file( $force = false ) {
 				return true;
 			}
 
+			$error = null;
 			if ( ! file_exists( $path . '/' . $dir ) ) {
 				if ( ! @mkdir( $path . '/' . $dir, 0755 ) ) { // @codingStandardsIgnoreLine
-					WC_Stripe_Logger::log( 'Error: ' . __( 'Unable to create domain association folder to domain root.', 'woocommerce-gateway-stripe' ) );
-					return false;
+					$error = __( 'Unable to create domain association folder to domain root.', 'woocommerce-gateway-stripe' );
 				}
 			}
-
 			if ( ! @copy( WC_STRIPE_PLUGIN_PATH . '/' . $file, $fullpath ) ) { // @codingStandardsIgnoreLine
-				WC_Stripe_Logger::log( 'Error: ' . __( 'Unable to copy domain association file to domain root.', 'woocommerce-gateway-stripe' ) );
-				return false;
+				$error = __( 'Unable to copy domain association file to domain root.', 'woocommerce-gateway-stripe' );
+			}
+
+			if ( isset( $error ) ) {
+				$url            = 'https://' . $_SERVER['HTTP_HOST'] . '/' . $dir . '/' . $file;
```

Comment:
How about we use [get_site_url()](https://developer.wordpress.org/reference/functions/get_site_url/) here?

```suggestion
				$url            = get_site_url( null, $dir . '/' . $file, 'https' );
```

Reply:
It's really hard to read that diff ^ so here's an easier-to-read version:

```php
$url            = get_site_url( null, $dir . '/' . $file, 'https' );
```

Reply:
I used the host directly based on my understanding that the file needs to be available from that path at the root level of the domain (as I believe is [described in this doc](https://stripe.com/docs/stripe-js/elements/payment-request-button#verifying-your-domain-with-apple-pay)) – rather than the site URL, which would be different for sites hosted under subdirectories. Actually, I'm just realizing now that would mean that the copied domain association file would be ineffective in this case to begin with, but that likely need not be dealt with in the scope of this issue.

Reply:
> the file needs to be available from that path at the root level of the domain

Ah, yeah you're right. I didn't think to take into account how my suggestion would work in multisites. Good point!